astropy/io/fits/hdu/compressed.py

```diff
@@ -374,6 +375,16 @@ def _remap_index(self, idx):
 
         return idx
 
+    def clear(self):
+        """
+        Remove all cards from the header.
+        """
+
+        self._table_header._cards = []
+        self._table_header._keyword_indices = collections.defaultdict(list)
+        self._table_header._rvkc_indices = collections.defaultdict(list)
```

Comment:
This should probably work ?
```suggestion
        self._table_header.clear()
```

Reply:
Yes, indeed, sorry for that... I guess I messed up in my debugging back and forth.